tests/std/tests/P0768R1_spaceship_operator/test.cpp

```diff
@@ -119,6 +121,69 @@ static_assert(test_common_type<std::strong_ordering, std::partial_ordering>());
 static_assert(test_common_type<std::strong_ordering, std::weak_ordering>());
 static_assert(test_common_type<std::strong_ordering, std::strong_ordering>());
 
+#if defined(__cpp_impl_three_way_comparison) && defined(__cpp_lib_concepts)
+constexpr auto my_cmp_three_way = [](const auto& left, const auto& right) { return left <=> right; };
+
+template <class Left, class Right>
+constexpr auto compare(const Left& left, const Right& right) {
+    using std::begin, std::end;
+    auto ret = std::lexicographical_compare_three_way(begin(left), end(left), begin(right), end(right));
+    auto ret2 =
+        std::lexicographical_compare_three_way(begin(left), end(left), begin(right), end(right), my_cmp_three_way);
+    assert(ret == ret2);
+    return ret;
+}
+
+template <class Ty1, class Ty2 = Ty1>
+constexpr bool test_algorithm2() {
+    std::array<Ty1, 4> original{0, 1, 2, 3};
+    std::array<Ty2, 4> same{0, 1, 2, 3}, lesser{0, 1, 1, 3}, greater{0, 1, 3, 3};
+    std::array<Ty2, 3> smaller{0, 1, 2}, smallerAndGreater{0, 2, 3};
+    std::array<Ty2, 5> bigger{0, 1, 2, 3, 4}, biggerAndLesser{0, 1, 2, 2, 4};

```

Comment:
We conventionally avoid declaring multiple variables on a single line (and have recently increased our consistency in product code). Our conventions for test code are somewhat more relaxed (e.g. I don't object to varying `const` placement), but I would still prefer to avoid multi-decls.